utils/interfaces/_library/miscs.py

```diff
@@ -109,9 +125,17 @@ def check(self, data):
             self.log_error(f"{data['log_filename']} content should be an array")
             return
 
+        span_types = []
+
         for trace in data["request"]["content"]:
             for span in trace:
-                if self.rid:
-                    if self.rid == _get_rid_from_span(span):
-                        self.log_debug(f"Found a trace for {self.message}")
+                if self.rid == _get_rid_from_span(span):
+                    self.log_debug(f"Found a trace for {self.message}")
+                    if self.span_type is None:  # no need to check for span type
                         self.set_status(True)
+                    else:  # check for the span type in all spans included in this trace
+                        span_types = [span.get("type") for span in trace]
+                        if self.span_type in span_types:
+                            self.set_status(True)
+                        else:
+                            self.log_error(f"Did not find {self.span_type} in {span_types}")
```

Comment:
Let give an even better error log:

```suggestion
                            self.log_error(f"Did not find span type '{self.span_type}' in reported span types: {span_types}")
```